cpp/src/io/csv/csv_reader.cu

```diff
@@ -207,6 +212,80 @@ std::string stringType(gdf_dtype dt){
 
 }
 
+/**
+ * reads contents of an arrow::io::RandomAccessFile in a char * buffer up to the number of bytes specified in bytes_to_read
+ * for non local filesystems where latency and availability can be an issue it will retry until it has exhausted its the read attemps and empty reads that are allowed
+ */
+gdf_error read_file_into_buffer(std::shared_ptr<arrow::io::RandomAccessFile> file, int64_t bytes_to_read, uint8_t* buffer, int total_read_attempts_allowed, int empty_reads_allowed){
+
+	if (bytes_to_read > 0){
+
+		int64_t total_read;
+		arrow::Status status = file->Read(bytes_to_read,&total_read, buffer);
+
+		if (!status.ok()){
+			return GDF_FILE_ERROR;
+		}
+
+		if (total_read < bytes_to_read){
+			//the following two variables shoudl be explained
+			//Certain file systems can timeout like hdfs or nfs,
+			//so we shoudl introduce the capacity to retry
+			int total_read_attempts = 0;
+			int empty_reads = 0;
+
+			while (total_read < bytes_to_read && total_read_attempts < total_read_attempts_allowed && empty_reads < empty_reads_allowed){
+				int64_t bytes_read;
+				status = file->Read(bytes_to_read-total_read,&bytes_read, buffer + total_read);
+				if (!status.ok()){
+					return GDF_FILE_ERROR;
+				}
+				if (bytes_read == 0){
+					empty_reads++;
+				}
+				total_read += bytes_read;
+			}
+			if (total_read < bytes_to_read){
+				return GDF_FILE_ERROR;
+			} else {
+				return GDF_SUCCESS;
+			}
+		} else {
+			return GDF_SUCCESS;
+		}
+	} else {
+		return GDF_SUCCESS;
+	}
+}
+
+
+/**
+ * @brief read in a CSV file
+ *
+ * Read in a CSV file, extract all fields, and return a GDF (array of gdf_columns) using arrow interface
+ **/
+
+gdf_error read_csv_arrow(csv_read_arg *args, std::shared_ptr<arrow::io::RandomAccessFile> arrow_file_handle)
```

Comment:
This is a C++ interface. In order to expose this via Python and other languages we will need to abstract it so that there are no C++ types in the interface. I recommend using a path or an opaque pointer in the external interface, and loading the file or casting into a RandomAccessFile internally.

Then it will be straightforward to expose this in Python.

Reply:
Python/Cython can use a C++ API directly.

Reply:
Jake that's beside the point. :) Note "other languages", e.g. Java. In any case, there's no reason to have this type in the signature, and after discussion there's no reason to add another API. Just use Arrow inside of `read_csv()`

Reply:
Hi  @harrism, we had a discussion on this subject with @williamBlazing  and we agree with you. Moreover, now in the new api of  `read_csv` you can send the buffer if `input_data_type` is `HOST_BUFFER`. When we started this PR the API didn't have this support. So we think that this PR no longer makes sense. We will close this PR unless there is another idea. 

`csv_params.filepath_or_buffer; /// If input_data_type is HOST_BUFFER, points to the host memory buffer
`
Cc @harrism.  @williamBlazing, @jrhemstad, @mrocklin